rules/library.bzl

```diff
@@ -267,10 +296,37 @@ def _xcframework_slice(*, xcframework_name, identifier, platform, platform_varia
             deps = [],
             tags = _MANUAL,
         )
+        imported_framework_name = _find_imported_framework_name(import_headers)
+        vfs_framework_name = imported_framework_name if imported_framework_name else xcframework_name
+        framework_vfs_overlay(
+            name = resolved_target_name_vfs_overlay,
+            framework_name = vfs_framework_name,
+            modulemap = import_module_map,
+            swiftmodules = import_swiftmodules,
+            hdrs = import_headers,
+            tags = _MANUAL,
+            extra_search_paths = xcframework_name,
+        )
     elif (linkage, packaging) == ("static", "library"):
+        # For a static library the `path` is the path to the library (`.a`) itself.
+        # We need the path to the top-level slice directory so we can find headers, modulemaps and swiftmodules.
+        slice_dir = paths.dirname(path)
+        import_headers, import_module_map, import_swiftmodules = _xcframework_slice_imports(slice_dir)
         native.objc_import(
             name = resolved_target_name,
             archives = [path],
+            hdrs = import_headers,
+            includes = ["%s/Headers" % slice_dir],
```

Comment:
Similar to #543 we need to provide `hdrs` and `includes` to support `#import <...>`